src/main/kotlin/app/revanced/patches/youtube/layout/returnyoutubedislike/patch/ReturnYouTubeDislikePatch.kt

```diff
@@ -137,35 +134,28 @@ class ReturnYouTubeDislikePatch : BytecodePatch(
 
         // region Hook for Short videos.
 
-        ShortsTextComponentParentFingerprint.result?.let {
-            context
-                .toMethodWalker(it.method)
-                .nextMethod(it.scanResult.patternScanResult!!.endIndex, true)
-                .getMethod().let { method ->
-                    with(method as MutableMethod) {
-                        // After walking, verify the found method is what's expected.
-                        if (returnType != ("Ljava/lang/CharSequence;") || parameterTypes.size != 1)
-                            return PatchResultError("Method signature did not match: $this $parameterTypes")
-
-                        val insertIndex = implementation!!.instructions.size - 1
-                        val spannedParameterRegister = instruction<OneRegisterInstruction>(insertIndex).registerA
-                        val parameter = instruction<BuilderInstruction35c>(insertIndex - 2).reference
-
-                        if (!parameter.toString().endsWith("Landroid/text/Spanned;"))
-                            return PatchResultError("Method signature parameter did not match: $parameter")
-
-                        insertShorts(insertIndex, spannedParameterRegister)
-                    }
-                }
-
-            // Additional hook, called after user dislikes.
-            with(it.mutableMethod) {
-                val insertIndex = it.scanResult.patternScanResult!!.startIndex + 2
-                val overwriteRegister = (implementation!!.instructions.elementAt(insertIndex - 1)
-                        as OneRegisterInstruction).registerA
-                insertShorts(insertIndex, overwriteRegister)
+        ShortsTextViewFingerprint.result?.let {
+            it.mutableMethod.apply {
+                val startIndex = it.scanResult.patternScanResult!!.startIndex
+                val endIndex = it.scanResult.patternScanResult!!.endIndex
+                val textViewFieldReference = // Obfuscated TextView field
+                    instruction<ReferenceInstruction>(endIndex - 2).reference
+                val isLikesBooleanReference = // Obfuscated boolean field
+                    instruction<ReferenceInstruction>(endIndex).reference
+
+                addInstructions(startIndex + 6, """
```

Comment:
Please give `startIndex + 6` a descriptive variable. 6 is a magic number.

Reply:
Maybe add a comment that explains why at this index. For example "Dislike view is moved into register at this index."